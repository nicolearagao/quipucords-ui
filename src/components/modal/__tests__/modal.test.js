import React from 'react';
import { Modal } from '../modal';

describe('Modal Component', () => {
  it('should render a basic component', () => {
    const props = {
      isOpen: true,
      disableFocusTrap: true
    };

    const component = renderComponent(<Modal {...props}>lorem ipsum</Modal>);
    expect(component.screen.render()).toMatchSnapshot('basic');
  });

  it('should pass specific and custom props', () => {
    const props = {
      backdrop: false
    };

    const component = renderComponent(<Modal {...props}>lorem ipsum</Modal>);
    expect(component.screen.render()).toMatchSnapshot('backdrop');

    const componentAriaLabel = component.setProps({ backdrop: true, 'aria-label': 'dolor sit' });
    expect(componentAriaLabel.props).toMatchSnapshot('aria-label');

    const componentContentOnly = component.setProps({ backdrop: true, isContentOnly: true });
    expect(componentContentOnly.props).toMatchSnapshot('isContentOnly');
  });

  it('should allow custom headers and footers', () => {
    // disableFocusTrap for testing only
    const props = {
      isOpen: true,
      disableFocusTrap: true
    };

    props.header = undefined;
    props.footer = undefined;
    const componentUndefined = renderComponent(<Modal {...props}>hello world</Modal>);
    expect(componentUndefined.screen.render()).toMatchSnapshot('undefined');
    componentUndefined.unmount();

    props.header = 'lorem ipsum';
    props.footer = 'dolor sit';
    const componentString = renderComponent(<Modal {...props}>hello world</Modal>);
    expect(componentString.screen.render()).toMatchSnapshot('string');
    componentString.unmount();

    props.header = () => 'lorem ipsum';
    props.footer = () => 'dolor sit';
    const componentFunction = renderComponent(<Modal {...props}>hello world</Modal>);
    expect(componentFunction.screen.render()).toMatchSnapshot('function');
    componentFunction.unmount();

    props.header = [<React.Fragment key="lorem">lorem ipsum</React.Fragment>];
    props.footer = [<React.Fragment key="dolor">dolor sit</React.Fragment>];
    const componentList = renderComponent(<Modal {...props}>hello world</Modal>);
    expect(componentList.screen.render()).toMatchSnapshot('list');
    componentList.unmount();

    props.header = <React.Fragment>lorem ipsum</React.Fragment>;
    props.footer = <React.Fragment key="dolor">dolor sit</React.Fragment>;
    const componentElement = renderComponent(<Modal {...props}>hello world</Modal>);
    expect(componentElement.screen.render()).toMatchSnapshot('element');
    componentElement.unmount();
  });
});
