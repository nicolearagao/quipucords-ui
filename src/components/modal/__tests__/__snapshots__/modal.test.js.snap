// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should allow custom headers and footers: element 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__rcue-width pf-c-backdrop__open"
  >
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-10"
            aria-label="t(modal.aria-label-default)"
            aria-labelledby="pf-modal-part-8"
            aria-modal="true"
            class="pf-c-modal-box pf-m-align-top"
            data-ouia-component-id="OUIA-Generated-Modal-9"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-8"
            role="dialog"
            style="--pf-c-modal-box--m-align-top--spacer: 5%;"
          >
            <header
              class="pf-c-modal-box__header"
            >
               lorem ipsum 
            </header>
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-10"
            >
              <div>
                 hello world 
              </div>
            </div>
            <footer
              class="pf-c-modal-box__footer"
            >
               dolor sit 
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;

exports[`Modal Component should allow custom headers and footers: function 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__rcue-width pf-c-backdrop__open"
  >
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-8"
            aria-label="t(modal.aria-label-default)"
            aria-labelledby="pf-modal-part-6"
            aria-modal="true"
            class="pf-c-modal-box pf-m-align-top"
            data-ouia-component-id="OUIA-Generated-Modal-7"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-6"
            role="dialog"
            style="--pf-c-modal-box--m-align-top--spacer: 5%;"
          >
            <header
              class="pf-c-modal-box__header"
            >
               lorem ipsum 
            </header>
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-8"
            >
              <div>
                 hello world 
              </div>
            </div>
            <footer
              class="pf-c-modal-box__footer"
            >
               dolor sit 
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;

exports[`Modal Component should allow custom headers and footers: list 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__rcue-width pf-c-backdrop__open"
  >
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-9"
            aria-label="t(modal.aria-label-default)"
            aria-labelledby="pf-modal-part-7"
            aria-modal="true"
            class="pf-c-modal-box pf-m-align-top"
            data-ouia-component-id="OUIA-Generated-Modal-8"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-7"
            role="dialog"
            style="--pf-c-modal-box--m-align-top--spacer: 5%;"
          >
            <header
              class="pf-c-modal-box__header"
            >
               lorem ipsum 
            </header>
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-9"
            >
              <div>
                 hello world 
              </div>
            </div>
            <footer
              class="pf-c-modal-box__footer"
            >
               dolor sit 
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;

exports[`Modal Component should allow custom headers and footers: string 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__rcue-width pf-c-backdrop__open"
  >
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-7"
            aria-label="t(modal.aria-label-default)"
            aria-labelledby="pf-modal-part-5"
            aria-modal="true"
            class="pf-c-modal-box pf-m-align-top"
            data-ouia-component-id="OUIA-Generated-Modal-6"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-5"
            role="dialog"
            style="--pf-c-modal-box--m-align-top--spacer: 5%;"
          >
            <header
              class="pf-c-modal-box__header"
            >
               lorem ipsum 
            </header>
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-7"
            >
              <div>
                 hello world 
              </div>
            </div>
            <footer
              class="pf-c-modal-box__footer"
            >
               dolor sit 
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;

exports[`Modal Component should allow custom headers and footers: undefined 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__rcue-width pf-c-backdrop__open"
  >
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-6"
            aria-label="t(modal.aria-label-default)"
            aria-labelledby="pf-modal-part-4"
            aria-modal="true"
            class="pf-c-modal-box pf-m-align-top"
            data-ouia-component-id="OUIA-Generated-Modal-5"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-4"
            role="dialog"
            style="--pf-c-modal-box--m-align-top--spacer: 5%;"
          >
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-6"
            >
              <div>
                 hello world 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;

exports[`Modal Component should pass specific and custom props: aria-label 1`] = `
{
  "aria-label": "dolor sit",
  "backdrop": true,
  "children": [
    {
      "displayName": undefined,
      "props": undefined,
      "type": undefined,
    },
  ],
  "className": null,
  "footer": null,
  "header": null,
  "isContentOnly": false,
  "position": "top",
  "positionOffset": "5%",
  "showClose": false,
  "t": [Function],
  "variant": null,
}
`;

exports[`Modal Component should pass specific and custom props: backdrop 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__hide-backdrop quipucords-modal__rcue-width"
  >
    <div />
  </div>
</screen>
`;

exports[`Modal Component should pass specific and custom props: isContentOnly 1`] = `
{
  "aria-label": null,
  "backdrop": true,
  "children": [
    {
      "displayName": undefined,
      "props": undefined,
      "type": undefined,
    },
  ],
  "className": null,
  "footer": null,
  "header": null,
  "isContentOnly": true,
  "position": "top",
  "positionOffset": "5%",
  "showClose": false,
  "t": [Function],
  "variant": null,
}
`;

exports[`Modal Component should render a basic component: basic 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__rcue-width pf-c-backdrop__open"
  >
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-2"
            aria-label="t(modal.aria-label-default)"
            aria-labelledby="pf-modal-part-0"
            aria-modal="true"
            class="pf-c-modal-box pf-m-align-top"
            data-ouia-component-id="OUIA-Generated-Modal-1"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-0"
            role="dialog"
            style="--pf-c-modal-box--m-align-top--spacer: 5%;"
          >
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-2"
            >
              <div>
                 lorem ipsum 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;
